code/game/objects/machinery/nuclearbomb.dm

```diff
@@ -242,98 +265,95 @@ obj/machinery/nuclearbomb/proc/nukehack_win(mob/user as mob)
-						auth = I
-		if (src.auth)
-			if (href_list["type"])
-				if (href_list["type"] == "E")
-					if (src.code == src.r_code)
-						src.yes_code = 1
-						src.code = null
-					else
-						src.code = "ERROR"
-				else
-					if (href_list["type"] == "R")
-						src.yes_code = 0
-						src.code = null
-					else
-						src.code += text("[]", href_list["type"])
-						if (length(src.code) > 5)
-							src.code = "ERROR"
-			if (src.yes_code)
-				if (href_list["time"])
-					var/time = text2num(href_list["time"])
-					src.timeleft += time
-					src.timeleft = min(max(round(src.timeleft), 60), 600)
-				if (href_list["timer"])
-					if (src.timing == -1.0)
-						return
-					if (src.safety)
-						to_chat(usr, "<span class='warning'>The safety is still on.</span>")
-						return
-					src.timing = !( src.timing )
-					if (src.timing)
-						if(!src.lighthack)
-							src.icon_state = "nuclearbomb2"
-						if(!src.safety)
-							GLOB.nukes_set_list |= src
-						else
-							GLOB.nukes_set_list -= src
-					else
-						GLOB.nukes_set_list -= src
-						if(!src.lighthack)
-							src.icon_state = "nuclearbomb1"
-				if (href_list["safety"])
-					src.safety = !( src.safety )
-					if(safety)
-						src.timing = 0
-						stop_processing()
-						GLOB.nukes_set_list -= src
-				if (href_list["anchor"])
-
-					if(removal_stage == 5)
-						src.anchored = FALSE
-						visible_message("<span class='warning'> \The [src] makes a highly unpleasant crunching noise. It looks like the anchoring bolts have been cut.</span>")
-						return
-
-					src.anchored = !( src.anchored )
-					if(src.anchored)
-						visible_message("<span class='warning'> With a steely snap, bolts slide out of [src] and anchor it to the flooring.</span>")
-					else
-						visible_message("<span class='warning'> The anchoring bolts slide back into the depths of [src].</span>")
-
-		for(var/mob/M in viewers(1, src))
-			if ((M.client && M.interactee == src))
-				src.attack_hand(M)
-	else
+	if (!(usr.contents.Find(src) || (in_range(src, usr) && istype(loc, /turf))))
 		usr << browse(null, "window=nuclearbomb")
 		return
-	return
 
+	usr.set_interaction(src)
+
+	if (href_list["auth"])
+		if (auth)
+			auth.loc = loc
+			yes_code = FALSE
+			auth = null
+		else
+			var/obj/item/I = usr.get_active_held_item()
+			if (istype(I, /obj/item/disk/nuclear))
+				if(usr.drop_held_item())
+					I.forceMove(src)
+					auth = I
+	if (auth)
+		if (href_list["type"])
+			if (href_list["type"] == "E")
+				if (code == r_code)
+					yes_code = TRUE
+					code = null
+				else
+					code = "ERROR"
+			else
+				if (href_list["type"] == "R")
+					yes_code = FALSE
+					code = null
+				else
+					code += text("[]", href_list["type"])
+					if (length(code) > 5)
+						code = "ERROR"
+		if (yes_code)
+			if (href_list["time"])
+				var/time = text2num(href_list["time"])
+				timeleft += time
+				timeleft = min(max(round(timeleft), 60), 600)
+			if (href_list["timer"])
+				if (timing == -1.0)
```

Comment:
```suggestion
				if (timing == -1)
```

Reply:
I need to test this, it was doing some weird shit when i moved to whole numbers before. 
The nuke inverts the number when turning off to save an off state but also keep the value.

Reply:
It shouldn't matter at all, which is why we're using whole numbers in 99% of the places